python/cudf/cudf/tests/test_buffer.py

```diff
@@ -40,19 +42,45 @@ def test_buffer_from_cuda_iface_contiguous(data):
 @pytest.mark.parametrize("dtype", ["uint8", "int8", "float32", "int32"])
 def test_buffer_from_cuda_iface_dtype(data, dtype):
     data = data.astype(dtype)
-    if dtype not in ("uint8", "int8"):
-        with pytest.raises(
-            TypeError, match="Buffer data must be of uint8 type"
-        ):
-            buf = Buffer(data=data, size=data.size)  # noqa: F841
-
-
-@pytest.mark.parametrize("size", [0, 1, 10, 100, 1000, 10_000])
-def test_buffer_copy(size):
-    data = cp.random.randint(low=0, high=100, size=size, dtype="u1")
-    buf = Buffer(data=data)
-    got = buf.copy()
-    assert got.size == buf.size
-    if size > 0:
-        assert got.ptr != buf.ptr
-    assert_array_equal(cp.asarray(buf), cp.asarray(got))
+    as_device_buffer_like(data)
```

Comment:
So something has changed here (previously `Buffer` only accepted `[u]int8` data?). Would be good to assert something about the returned value (although this was not done previously)

Reply:
Updated `test_buffer_from_cuda_iface_dtype()` to check against a flatten cupy array.